### ISSUE TITLE
Behavior change moving from pgtype.TextArray to pgtype.Array[string] while using stdlib

### ISSUE THREAD

**Issue by oschwald** [jackc/pgx#1312]
**Describe the bug**

When upgrading stdlib code from using `pgtype.TextArray` to `pgtype.Array[string]`, `NULL` values can no longer be scanned. 

**To Reproduce**

If possible, please provide runnable example such as:

```go
package main

import (
	"database/sql"
	"log"
	"os"

	"github.com/jackc/pgx/v5/pgtype"
)

func main() {
	db, err := sql.Open("pgx", , os.Getenv("DATABASE_URL"))
	if err != nil {
		log.Fatal(err)
	}

	var arr pgtype.Array[string]
	m := pgtype.NewMap()
	err = db.QueryRow(
		`SELECT NULL::text[]`,
	).Scan(m.SQLScanner(&arr))
	if err != nil {
		log.Fatal(err)
	}
}
```

**Expected behavior**

I would have expected no error but `arr.Valid` to be false.

**Actual behavior**

```
sql: Scan error on column index 0, name "text": invalid array: strconv.ParseInt: parsing "": invalid syntax
```

**Version**
 - Go: `go version go1.19.1 linux/amd64`
 - PostgreSQL: `PostgreSQL 14.5 (Ubuntu 14.5-1.pgdg22.04+1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 11.2.0-19ubuntu1) 11.2.0, 64-bit`
 - pgx: `5.0.0`

**Comment by jackc** [comment]
Thanks for the clear report. Fixed in 335c8621ff0daa87e8602c8ae18c55fb9da81352.

**Comment by oschwald** [comment]
Thank you! The fix resolved the issue for us.
